eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/UnifiedMap.java

```diff
@@ -1765,6 +1765,42 @@ else if (this.table[i] != null)
     }
 
     @Override
+    public Optional<Pair<K, V>> detectOptional(Predicate2<? super K, ? super V> predicate)
+    {
+        for (int i = 0; i < this.table.length; i += 2)
+        {
+            if (this.table[i] == CHAINED_KEY)
+            {
+                Object[] chainedTable = (Object[]) this.table[i + 1];
+                for (int j = 0; j < chainedTable.length; j += 2)
+                {
+                    if (chainedTable[j] != null)
+                    {
+                        K key = this.nonSentinel(chainedTable[j]);
+                        V value = (V) chainedTable[j + 1];
+                        if (predicate.accept(key, value))
+                        {
+                            return Optional.of(Tuples.pair(key, value));
+                        }
+                    }
+                }
+            }
+            else if (this.table[i] != null)
+            {
+                K key = this.nonSentinel(this.table[i]);
+                V value = (V) this.table[i + 1];
+
+                if (predicate.accept(key, value))
+                {
+                    return Optional.of(Tuples.pair(key, value));
+                }
+            }
+        }
+
+        return Optional.empty();
+    }
+
+     @Override
```

Comment:
There's extra whitespace on this last line. I figured the CheckStyle build would catch that?

Reply:
Great catch indeed! 

Interesting why the build hasn't failed... 🤔

Reply:
Double checked checkstyle build locally, it doesn't fail either. 

Let me look into this separately why CheckStyle doesn't detect the extra white space...

I'll fix the white space itself soon.